--- a/mt7915/init.c
+++ b/mt7915/init.c
@@ -289,7 +289,7 @@ static void __mt7915_init_txpower(struct
 	int pwr_delta = mt7915_eeprom_get_power_delta(dev, sband->band);
 	struct mt76_power_limits limits;
 
-	phy->sku_limit_en = true;
+	phy->sku_limit_en = false;
 	phy->sku_path_en = true;
 	for (i = 0; i < sband->n_channels; i++) {
 		struct ieee80211_channel *chan = &sband->channels[i];
@@ -314,8 +314,9 @@ static void __mt7915_init_txpower(struct
 
 		target_power += path_delta;
 		target_power = DIV_ROUND_UP(target_power, 2);
-		chan->max_power = min_t(int, chan->max_reg_power,
-					target_power);
+
+		target_power = 30;
+		chan->max_power = target_power;
 		chan->orig_mpwr = target_power;
 	}
 }
